### PR TITLE
Replace `tail.nonEmpty` with `sizeIs > 1` as sources might be `Nil`

### LINKED ISSUE #8000: Partest should fail nice on empty dir
On an empty dir, instead of a nice error or ignoring it,

```
java.lang.RuntimeException: Error running /home/apm/clones/scala/test/files/pos/AAAA
	at scala.tools.partest.nest.SuiteRunner.liftedTree1$1(Runner.scala:773)
	at scala.tools.partest.nest.SuiteRunner.runTest(Runner.scala:771)
	at scala.tools.partest.nest.SuiteRunner$$anonfun$19$$anonfun$apply$6.apply(Runner.scala:786)
	at scala.tools.partest.nest.SuiteRunner$$anonfun$19$$anonfun$apply$6.apply(Runner.scala:786)
	at scala.tools.partest.package$$anon$2.call(package.scala:133)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:724)
Caused by: java.lang.UnsupportedOperationException: tail of empty list
```

### PR DESCRIPTION
**Pull request by Philippus**
fixes https://github.com/scala/bug/issues/8000

### CODE REVIEW

File: src/partest/scala/tools/partest/nest/Runner.scala [hunk 1]
```diff
@@ -437,7 +437,7 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
 
   /** Grouped files in group order, and lex order within each group. */
   def groupedFiles(sources: List[File]): List[List[File]] = (
-    if (sources.tail.nonEmpty) {
+    if (sources.size > 1) {
```
Comment (Ichoran) [inline src/partest/scala/tools/partest/nest/Runner.scala]: This is the wrong fix--`.size` is an O(n) operation on a list and this doesn't require that.  Instead, use `size_compare` or `sources.drop(1).nonEmpty` (both are cheap).

**Review by Ichoran — DISMISSED** [review]

File: src/partest/scala/tools/partest/nest/Runner.scala (hunk 1 above)
Comment (NthPortal) [reply src/partest/scala/tools/partest/nest/Runner.scala]: or `sizeIs > 1`

Comment (som-snytt) [reply src/partest/scala/tools/partest/nest/Runner.scala]: What crafty magic!

**Comment by som-snytt** [comment]
Thanks, that was such a long time ago. But I remember soc complaining that the psychologically significant (in those simpler days) threshold of issue 8000 "wasn't even a real bug." Thanks for proving him wrong. I'm glad the fix waited to be solved using the advanced technology available to us in our own time.

**Review by NthPortal — APPROVED** [review]

**Pull request merged**